source/standalone/tutorials/04_sensors/benchmark_cameras.py

```diff
@@ -0,0 +1,621 @@
+parser.add_argument(
+    "--ray_caster_visible_mesh_prim_paths",
+    nargs="+",
+    type=str,
+    default=["/World/ground"],
+    help="WARNING: Ray Caster can currently only cast against a single, static, object",
+)
+
+parser.add_argument(
+    "--convert_depth_to_camera_to_image_plane",
+    action="store_true",
+    default=True,
+    help=(
+        "Enable undistorting from perspective view (distance to camera replicator)"
+        "to orthogonal view (distance to plane replicator) for depth."
+    ),
+)
+
+parser.add_argument(
+    "--keep_raw_depth",
+    dest="convert_depth_to_camera_to_image_plane",
+    action="store_false",
+    help=(
+        "Disable undistorting from perspective view (distance to camera)"
+        "to orthogonal view (distance to plane replicator) for depth."
+    ),
+)
+
+parser.add_argument(
+    "--height",
+    type=int,
+    default=120,
+    required=False,
+    help="Height in pixels of cameras",
+)
+
+parser.add_argument(
+    "--width",
+    type=int,
+    default=140,
+    required=False,
+    help="Width in pixels of cameras",
+)
+
+parser.add_argument(
+    "--warm_start_length",
+    type=int,
+    default=3,
+    required=False,
+    help="How many steps to run the sim before starting benchmark",
+)
+
+parser.add_argument(
+    "--num_objects", type=int, default=10, required=False, help="How many objects to spawn into the scene."
+)
+
+parser.add_argument(
+    "--experiment_length",
+    type=int,
+    default=30,
+    required=False,
+    help="How many steps to average over",
+)
+
+AppLauncher.add_app_launcher_args(parser)
+args_cli = parser.parse_args()
+args_cli.enable_cameras = True
+
+if args_cli.visualize:
+    print("[WARNING]: You have selected to visualize. which means your benchmark results will not be meaningful.")
+    import matplotlib
+
+    matplotlib.use("Agg")  # Use a non-interactive backend
+
+if len(args_cli.ray_caster_visible_mesh_prim_paths) > 1:
+    print("[WARNING]: Ray Casting is only currently supported for a single, static object")
+# launch omniverse app
+app_launcher = AppLauncher(args_cli)
+simulation_app = app_launcher.app
+
+"""Rest everything follows."""
+
+import numpy as np
+import open3d as o3d
+import random
+import time
+import torch
+from matplotlib import pyplot as plt
+
+import omni.isaac.core.utils.prims as prim_utils
+
+import omni.isaac.lab.sim as sim_utils
+from omni.isaac.lab.assets import RigidObject, RigidObjectCfg
+from omni.isaac.lab.sensors.camera import Camera, CameraCfg, TiledCamera, TiledCameraCfg
+from omni.isaac.lab.sensors.ray_caster import RayCasterCamera, RayCasterCameraCfg, patterns
+from omni.isaac.lab.utils.math import convert_perspective_depth_image_to_orthogonal_depth_image, unproject_depth
+
+
+def create_camera_base(
+    camera_cls: type[Camera | TiledCamera],
+    camera_cfg: type[CameraCfg | TiledCameraCfg],
+    num_cams: int,
+    data_types: list[str],
+    height: int,
+    width: int,
+) -> Camera | TiledCamera | None:
```

Comment:
NIT: Try to follow the google python docstring style guide with Args, Returns, Raises for all of these methods: https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html